$OpenBSD$
- OpenBSD doesn't need libdl
- use pkg-config to find glib libs
- don't attempt to overwrite an optimized library with a non-existant one
--- c-lib/Makefile.orig	Fri Aug  3 12:44:51 2012
+++ c-lib/Makefile	Sun Mar 24 12:32:34 2013
@@ -9,7 +9,7 @@ PROG_CFLAGS = -Wall
 OCAML_LIBDIR := $(shell ocamlc -where)
 INCLUDES = -I . -I $(OCAML_LIBDIR) $(shell pkg-config --cflags glib-2.0)
 CUDF_LDFLAGS = -L$(OCAML_LIBDIR) -L$(CURDIR)
-CUDF_CLIBS = -lcudf -lm -ldl -lunix -lncurses -lglib-2.0
+CUDF_CLIBS = -lcudf -lm -lunix -lncurses $(shell pkg-config --libs glib-2.0)
 OCAMLC = ocamlfind ocamlc -package unix,extlib
 OCAMLOPT = ocamlfind ocamlopt -package unix,extlib
 
@@ -68,7 +68,6 @@ install: cudf.pc
 	else \
 		cp libcudf.a $(DESTDIR)/$(LIBDIR)/libcudf.a ; \
 	fi
-	cp libcudf.a $(DESTDIR)/$(LIBDIR)
 	cp cudf.h $(DESTDIR)/$(INCDIR)
 	cp cudf.pc $(DESTDIR)/$(PCDIR)
 
